File: src/flydocs/interfaces/enums/field_type.py
# Copyright 2024-2026 Firefly Software Foundation
#
# Licensed under the Apache License, Version 2.0 (the "License");
# you may not use this file except in compliance with the License.
# You may obtain a copy of the License at
#
#     http://www.apache.org/licenses/LICENSE-2.0
#
# Unless required by applicable law or agreed to in writing, software
# distributed under the License is distributed on an "AS IS" BASIS,
# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
# See the License for the specific language governing permissions and
# limitations under the License.

"""Supported field primitives + standard formats for the public extraction schema."""

from __future__ import annotations

from enum import StrEnum


class FieldType(StrEnum):
    """JSON-Schema-aligned primitive set for the public Field model."""

    STRING = "string"
    NUMBER = "number"
    INTEGER = "integer"
    BOOLEAN = "boolean"
    ARRAY = "array"
    OBJECT = "object"


class StandardFormat(StrEnum):
    """Standard format hints applied to typed field values at validation time."""

    DATE = "date"
    DATE_TIME = "date-time"
    TIME = "time"
    EMAIL = "email"
    URI = "uri"
    UUID = "uuid"
    CURRENCY = "currency"

File: sdks/java/src/main/java/com/firefly/flydocs/sdk/model/FieldType.java
/*
 * Copyright 2026 Firefly Software Solutions Inc
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package com.firefly.flydocs.sdk.model;

import com.fasterxml.jackson.annotation.JsonValue;

/**
 * Supported primitives for a {@link FieldSpec}. Mirrors the service-side
 * {@code FieldType} enum; the wire form is the lowercase value
 * ({@code "string"}, {@code "number"}, ...).
 */
public enum FieldType {
    STRING("string"),
    NUMBER("number"),
    INTEGER("integer"),
    BOOLEAN("boolean"),
    ARRAY("array");

    private final String wire;

    FieldType(String wire) {
        this.wire = wire;
    }

    /** JSON wire value. */
    @JsonValue
    public String wire() {
        return wire;
    }
}
